maths/factorial.dart

```diff
@@ -2,10 +2,14 @@ void main() {
   var n = 5;
   var fac = factorial(n);
   print("$n! = $fac"); /* output: 5! = 120 */
+
 }
 
 /* calculate factorial of n*/
-int factorial(var n) {
+int factorial(int n) {
+
+  assert(n >= 0);
+
```

Comment:
Heyy, Can we improve this file to have tests. like we are having for the newer files. Since we are fixing the bugs. let's add a few test cases.